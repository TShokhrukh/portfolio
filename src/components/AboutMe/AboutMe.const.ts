import { cn } from '@bem-react/classname'

export const cnAboutMe = cn('AboutMe')
