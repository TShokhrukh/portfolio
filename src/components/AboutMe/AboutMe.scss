.AboutMe {
  --about-me-template-columns: 1fr;
  --about-gap: #{$spacer * 2};

  display: grid;
  grid-template-columns: var(--about-me-template-columns);
  gap: var(--about-gap);
}

@include media-breakpoint-up(xl) {
  .AboutMe {
    --about-me-template-columns: 40% 1fr;
  }
}