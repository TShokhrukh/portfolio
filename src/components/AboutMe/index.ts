export * from './build'
export * from './__types__'
export { AboutMe as AboutMeBase } from './AboutMe'
