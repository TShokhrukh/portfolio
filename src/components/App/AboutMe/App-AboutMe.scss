.App-AboutMe {
  padding: var(--app-section-padding-y) var(--app-section-padding-x);
}