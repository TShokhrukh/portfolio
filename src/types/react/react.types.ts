import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
