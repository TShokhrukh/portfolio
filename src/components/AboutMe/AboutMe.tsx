import React, { FC } from 'react'
import { AboutMeAvatar } from './Avatar'
import { AboutMeDesc } from './Desc'
import { cnAboutMe } from './AboutMe.const'
import { Context } from './lib/context'
import type { IAboutMeProps } from './__types__'
import avatar from './res/avatar.png'
import './AboutMe.scss'

export const AboutMe: FC<IAboutMeProps> = ({
  component: Component = 'div',
  className,
  ...props
}) => {
  return (
    <Context.Provider value={{ avatar }}>
      <Component {...props} className={cnAboutMe({}, [className])}>
        <AboutMeAvatar />
        <AboutMeDesc />
      </Component>
    </Context.Provider>
  )
}
