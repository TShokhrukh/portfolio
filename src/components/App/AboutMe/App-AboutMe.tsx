import React, { FC } from 'react'
import { Container } from '~/components/Container'
import { AboutMe } from '~/components/AboutMe'
import { cnApp } from '../App.const'

import './App-AboutMe.scss'

export const AppAboutMe: FC = () => {
  return (
    <section id='about_me' className={cnApp('AboutMe', {})}>
      <Container>
        <AboutMe />
      </Container>
    </section>
  )
}
