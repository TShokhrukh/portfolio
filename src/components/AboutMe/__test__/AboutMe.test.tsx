/* global test, jest, expect, describe */
import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { AboutMe } from '../AboutMe'

describe('AboutMe', () => {
  test('must return the component template', () => {
    render(<AboutMe data-testid='about_me' />)

    expect(screen.getByTestId('about_me')).toMatchSnapshot()
  })
})
