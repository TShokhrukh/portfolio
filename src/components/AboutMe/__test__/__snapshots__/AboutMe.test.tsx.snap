// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe must return the component template 1`] = `
<div
  class="AboutMe"
  data-testid="about_me"
>
  <img
    alt="..."
    class="Img AboutMe-Avatar"
  />
  <div
    class="AboutMe-Desc"
  >
    <div>
      <span
        class="Badge Badge_color_purple"
      >
        aboutMe.aboutMe
      </span>
    </div>
    <h2
      class="Typography Typography_color_white Typography_variant_h2 AboutMe-Name"
    >
      aboutMe.me
    </h2>
    <p
      class="Typography"
    >
      <span>
        aboutMe.desc
      </span>
    </p>
  </div>
</div>
`;
